fdbserver/storageserver.actor.cpp

```diff
@@ -2209,10 +2224,24 @@ ACTOR Future<Version> watchWaitForValueChange(StorageServer* data, SpanContext p
 
 	state Version minVersion = data->data().latestVersion;
 	state Future<Void> watchFuture = data->watches.onChange(metadata->key);
+	if (tenantId != TenantInfo::INVALID_TENANT) {
+		watchFuture = watchFuture || data->tenantWatches.onChange(tenantId);
+	}
 	state ReadOptions options;
+	state bool firstPass = true;
 	loop {
 		try {
-			metadata = data->getWatchMetadata(key);
+			if (tenantId != TenantInfo::INVALID_TENANT) {
```

Comment:
Since you haven't changed `watchValueWaitForVersion`, I think we can actually just throw `tenant_removed` unconditionally here. And since what we're doing doesn't quite line up with `checkTenantEntry`, it may be cleaner to check the map directly:

```
auto view = data->tenantMap.at(latestVersion);
if (view.find(tenantId) == view.end()) {
    throw tenant_removed();
}
```